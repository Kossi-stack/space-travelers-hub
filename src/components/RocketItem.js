import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import {
  addRocketReservation,
  cancelRocketReservation,
} from '../redux/rockets/rockets';

const RocketItem = (props) => {
  const { rocketProps } = props;
  const dispatch = useDispatch();
  // eslint-disable-next-line max-len

  const handleAddReservation = (e) => {
    e.preventDefault();
    dispatch(addRocketReservation(rocketProps.id));
  };

  const handleCancelReservation = (e) => {
    e.preventDefault();
    dispatch(cancelRocketReservation(rocketProps.id));
  };

  return (
    <li className="rockets-item" key={rocketProps.id}>
      <img
        alt="img"
        src={rocketProps.flickr_images[0]}
        className="rockets-item__img"
      />
      <div className="rockets-item__desc">
        <h3 className="rockets-item__desc-h3">{rocketProps.rocket_name}</h3>
        <p className="rockets-item__desc-p">
          {rocketProps.reserved ? (
            <span className="badge badge-active spacing-right">Reserved</span>
          ) : (
            ''
          )}
          {rocketProps.description}
        </p>

        {rocketProps.reserved && (
          <button
            className="btn-gray"
            type="button"
            onClick={handleCancelReservation}
          >
            Cancel Reservation
          </button>
        )}

        {!rocketProps.reserved && (
          <button className="btn" type="button" onClick={handleAddReservation}>
            Reserve Rocket
          </button>
        )}
      </div>
    </li>
  );
};

RocketItem.propTypes = {
  rocketProps: PropTypes.shape({
    id: PropTypes.number.isRequired,
    rocket_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

export default RocketItem;
