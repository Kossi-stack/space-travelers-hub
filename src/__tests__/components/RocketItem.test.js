import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import RocketItem from '../../components/RocketItem';

it('renders rocket item correctly', () => {
  const rocket = {
    id: 1,
    rocket_name: 'falcon 1',
    description: 'some description',
    reserved: true,
    flickr_images: [
      'https://imgur.com/DaCfMsj.jpg',
      'https://imgur.com/azYafd8.jpg',
    ],
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <RocketItem key={rocket.id} rocketProps={rocket} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
