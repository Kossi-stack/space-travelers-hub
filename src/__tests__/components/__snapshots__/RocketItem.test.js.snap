// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rocket item correctly 1`] = `
<li
  className="rockets-item"
>
  <img
    alt="img"
    className="rockets-item__img"
    src="https://imgur.com/DaCfMsj.jpg"
  />
  <div
    className="rockets-item__desc"
  >
    <h3
      className="rockets-item__desc-h3"
    >
      falcon 1
    </h3>
    <p
      className="rockets-item__desc-p"
    >
      <span
        className="badge badge-active spacing-right"
      >
        Reserved
      </span>
      some description
    </p>
    <button
      className="btn-gray"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation
    </button>
  </div>
</li>
`;
